app/frontend/src/components/forms/submission/ManageSubmissionUsers.vue

```diff
@@ -0,0 +1,303 @@
+      // search box
+      findUsers: null,
+      isLoadingDropdown: false,
+      userSearchResults: [],
+      userSearchSelection: null,
+    };
+  },
+  methods: {
+    ...mapActions('notifications', ['addNotification']),
+    // show users in dropdown that have a text match on multiple properties
+    async addUser() {
+      if (this.userSearchSelection) {
+        const id = this.userSearchSelection.id;
+        if (this.userTableList.some((u) => u.id === id)) {
+          this.addNotification({
+            type: 'warning',
+            message: `User ${this.userSearchSelection.username} is already in the list of team members.`,
+          });
+        } else {
+          this.modifyPermissions(id, [
+            FormPermissions.SUBMISSION_UPDATE,
+            FormPermissions.SUBMISSION_READ,
+          ]);
+        }
+      }
+      // reset search field
+      this.userSearchSelection = null;
+    },
+    filterObject(item, queryText) {
+      // eslint-disable-next-line
+      for (const [key, value] of Object.entries(item)) {
+        if (
+          value !== null &&
+          value.toLocaleLowerCase().includes(queryText.toLocaleLowerCase())
+        ) {
+          return true;
+        }
+      }
+    },
+    async getSubmissionUsers() {
+      this.isLoadingTable = true;
+      try {
+        const response = await rbacService.getSubmissionUsers({
+          formSubmissionId: this.submissionId,
+        });
+        if (response.data) {
+          this.userTableList = this.transformResponseToTable(response.data);
+        }
+      } catch (error) {
+        this.addNotification({
+          message:
+            'An error occured while trying to fetch users for this submission.',
+          consoleError: `Error getting users for ${this.submissionId}: ${error}`,
+        });
+      } finally {
+        this.isLoadingTable = false;
+      }
+    },
+    async modifyPermissions(userId, permissions) {
+      this.isLoadingTable = true;
+      try {
+        // Add the selected user with read/update permissions on this submission
+        const response = await rbacService.setSubmissionUserPermissions(
+          { permissions: permissions },
+          {
+            formSubmissionId: this.submissionId,
+            userId: userId,
+          }
+        );
+        if (response.data) {
+          this.userTableList = this.transformResponseToTable(response.data);
+        }
+      } catch (error) {
+        this.addNotification({
+          message:
+            'An error occured while trying to update users for this submission.',
+          consoleError: `Error setting user permissions. Sub: ${this.submissionId} User: ${userId} Error: ${error}`,
+        });
+      } finally {
+        this.isLoadingTable = false;
+      }
+    },
+    removeUser(userRow) {
+      this.userToDelete = userRow;
+      this.showDeleteDialog = true;
+    },
+    transformResponseToTable(responseData) {
+      return responseData
+        .map((su) => {
```

Comment:
could we make the su variable into something more descriptive?

Reply:
We use 1 or 2 characters pretty much everywhere as our `map` etc iterators. I think `su` works here (Submission User)?

Reply:
I didn't quite pick up that su meant submission user, but in most cases, we don't actually care what the iterated element is. You could easily just say "e" or "element" and you'd be fine since it's a map lambda function.